app/controllers/profile_controller.rb

```diff
@@ -6,5 +6,32 @@ def require_user
     render file: "/public/404" unless current_user?
   end
 
-  def index; end
+  def index
+    @user = current_user
+  end
+
+  def edit
+    @user = current_user
+  end
+
+  def update
+    @user = current_user
+    @user.update(user_params)
+    if User.where(email:@user.email).where.not(id:@user.id) != []
```

Comment:
Def some opportunities to refactor here, we can take a look at the user-login branch for examples